Internal/Vehicle/VehicleNext.as

```diff
@@ -250,6 +250,31 @@ namespace VehicleState
 		return Dev::GetOffsetInt32(vis, g_offsetCruiseDisplaySpeed);
 	}
 
+	// Get the current vehicle type.
+	VehicleType GetVehicleType(CSceneVehicleVisState@ vis)
+	{
+		if (g_offsetVehicleType == 0) {
+			auto type = Reflection::GetType("CSceneVehicleVisState");
+			if (type is null) {
+				error("Unable to find reflection info for CSceneVehicleVisState!");
+				return VehicleType::Stadium;
+			}
+			g_offsetVehicleType = type.GetMember("InputSteer").Offset - 8;
+		}
+
+		CTrackMania@ App = cast<CTrackMania@>(GetApp());
+		if (App.RootMap !is null) {
+			if (App.RootMap.VehicleName.GetName() == "CarSnow")
+				return VehicleType::Snow;
+			// if (map.VehicleName.GetName() == "CarRally")  // to update when car is added
+			// 	return VehicleType::Rally;
+			// if (map.VehicleName.GetName() == "CarDesert")  // to update when car is added
+			// 	return VehicleType::Desert;
+		}
```

Comment:
Why are we checking the vehicle on the map? What if the vehicle is actually different than what the map says?

Reply:
If the map sets the vehicle to something that's not stadium car, it's currently impossible to switch from whatever is set there

Reply:
Okay, but I don't think we should rely on that when the value is already available to us anyway, right?

Reply:
Sure it's available, but it makes it a lot easier for a developer to use this one function to return the vehicle type rather than checking the map's vehicle themselves, then using this function anyway if the map doesn't change it from stadium

Reply:
For reference, if the map changes the vehicle, the value in the vis state still changes but would be incorrect

Reply:
outdated and no longer relevant since rally update